feat(storybook): remove hasIcon/Unit props

bar-tay opened the pull request:
closes #1028

thrbnhrtmnn reviewed (changes requested):
Hey @bar-tay ,

I see some issues in Storybook that need to be fixed:

Button Text: If icon = undefined, the button size changes and the button looks not as defined in design (happens both with trailing and leading iconPosition)

Input Field Text: The icon stil is shown, even if icon = undefined

Input Field Number: If unit = undefined, the spacing between placeholder/value and unit should not be shown. Changing between suffix and prefix therefor should not have an effect, when unit = undefined.

bar-tay commented:
> Input Field Number: If unit = undefined, the spacing between placeholder/value and unit should not be shown. Changing between suffix and prefix therefor should not have an effect, when unit = undefined.

I implemented the suggestions. For the Number Input, I changed it back so the unitPosition only shows up when a unit was set. Since the position is only relevant if a unit was set.

thrbnhrtmnn reviewed (approved):
Button Text, Input Field Text and Input Field Number now look as desired in Storybook. Thanks for the update @bar-tay !

angsherpa456 commented on packages/icons/icons-optimized/index.ts:
```diff
@@ -359,6 +359,7 @@ const blrSelectionOnXs = () => import(`!!raw-loader!@boiler/icons/icons-optimize
 const blrSelectionOnXxs = () => import(`!!raw-loader!@boiler/icons/icons-optimized/${'Selection_on_xxs.svg'}`);
 
 export const PureIconKeys = [
+  undefined,
```
Rather than adding undefined here, did you try like the following which is currently applied approach. Having undefined as an icon in the PureIconKeys looks quite off.
 
icon: {
      description: 'Select the icon of the component.',
      options: **[undefined, ...PureIconKeys]**,
      control: { type: 'select' },
      table: {
        category: 'Content / Settings',
      },
    }

bar-tay replied:
Done

bar-tay commented on packages/ui-library/src/components/input-field-number/index.stories.ts:
```diff
@@ -138,18 +141,7 @@ export default {
-        category: 'Content / Settings',
-      },
-      if: { arg: 'hasUnit', eq: true },
+      if: { arg: 'unit', neq: undefined },
```
The neq operator stands for "not equal." in Storybook.

angsherpa456 replied:
I think you can completely remove the if statement because by the description it means that the unit will always be present just that the value might be either "prefix or suffix".

bar-tay replied:
I changed it so undefined is not a value of the Units array anymore. Units can be undefined.

angsherpa456 replied:
Aah, makes sense, please ignore it then. Still the pipeline is failing, please fix that and should be good.

bar-tay replied:
done
